src/mpDris2.in.py

```diff
@@ -578,27 +585,73 @@ def update_metadata(self):
                 logger.error("Can't cast value %r to %s" %
                              (value, allowed_tags[key]))
 
-    def notify_about_track(self, meta, state='play'):
-        uri = 'sound'
-        if 'mpris:artUrl' in meta:
-            uri = meta['mpris:artUrl']
+    def convertTimestamp(self, secs, micros=0):
+        seconds, minutes, hours = 0, 0, 0
 
-        title = 'Unknown Title'
-        if 'xesam:title' in meta:
-            title = meta['xesam:title']
-        elif 'xesam:url' in meta:
-            title = meta['xesam:url'].split('/')[-1]
+        if micros > 0:
+            secs += micros / 1000000
+        if secs > 0:
+            seconds = int(secs % 60)
+            minutes = int((secs / 60) % 60)
+            hours = int(secs / 3600)
 
-        artist = 'Unknown Artist'
-        if 'xesam:artist' in meta:
-            artist = ", ".join(meta['xesam:artist'])
+        if hours == 0:
+            duration = "{}:{:0>2}".format(minutes, seconds)
+        else:
+            duration = "{}:{}:{:0>2}".format(hours, minutes, seconds)
+
+        return duration
+
+    def format_notification(self, meta, text):
+        """format '%property%' in a string for it's actual value"""
+
+        format_strings = {
+            "album": meta.get("xesam:album", "Unknown Album"),
+            "title": meta.get("xesam:title", "Unknown Title"),
+            "id": meta.get("mpris:trackid", "").split("/")[-1],
+            "time": self.convertTimestamp(0, meta.get("mpris:length", 0)),
```

Comment:
I think this should be:

```suggestion
            "time": self.convertTimestamp(meta.get("mpris:length", 0), 0),
```

Reply:
No, mpris:length returns the length of the current track in microseconds. It must be the second argument.